feature_engine/outliers/winsorizer.py

```diff
@@ -136,108 +246,28 @@ def __init__(
-        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
-
-        if self.missing_values == "raise":
-            # check if dataset contains na
-            _check_contains_na(X, self.variables_)
-            _check_contains_inf(X, self.variables_)
-
-        self.right_tail_caps_ = {}
-        self.left_tail_caps_ = {}
-
-        # estimate the end values
-        if self.tail in ["right", "both"]:
-            if self.capping_method == "gaussian":
-                self.right_tail_caps_ = (
-                    X[self.variables_].mean() + self.fold * X[self.variables_].std()
-                ).to_dict()
-
-            elif self.capping_method == "iqr":
-                IQR = X[self.variables_].quantile(0.75) - X[self.variables_].quantile(
-                    0.25
-                )
-                self.right_tail_caps_ = (
-                    X[self.variables_].quantile(0.75) + (IQR * self.fold)
-                ).to_dict()
-
-            elif self.capping_method == "quantiles":
-                self.right_tail_caps_ = (
-                    X[self.variables_].quantile(1 - self.fold).to_dict()
-                )
-
-        if self.tail in ["left", "both"]:
-            if self.capping_method == "gaussian":
-                self.left_tail_caps_ = (
-                    X[self.variables_].mean() - self.fold * X[self.variables_].std()
-                ).to_dict()
-
-            elif self.capping_method == "iqr":
-                IQR = X[self.variables_].quantile(0.75) - X[self.variables_].quantile(
-                    0.25
-                )
-                self.left_tail_caps_ = (
-                    X[self.variables_].quantile(0.25) - (IQR * self.fold)
-                ).to_dict()
-
-            elif self.capping_method == "quantiles":
-                self.left_tail_caps_ = X[self.variables_].quantile(self.fold).to_dict()
-
-        self.n_features_in_ = X.shape[1]
-
-        return self
+        super().__init__(capping_method, tail, fold, variables, missing_values)
+        self.add_indicators = add_indicators
 
     # Ugly work around to import the docstring for Sphinx, otherwise not necessary
     def transform(self, X: pd.DataFrame) -> pd.DataFrame:
-        X = super().transform(X)
-
-        return X
+        X = _is_dataframe(X)
+        X_out = super().transform(X)
+        if self.add_indicators:
+            if self.tail in ["left", "both"]:
+                X_low = X_out > X
+                X_low.columns = [str(cl) + "_low" for cl in X.columns.values]
+            if self.tail in ["right", "both"]:
+                X_high = X_out < X
+                X_high.columns = [str(cl) + "_high" for cl in X.columns.values]
+            if self.tail in ["left", "both"]:
+                X_out = pd.concat([X_out, X_low], axis=1)
+            if self.tail in ["right", "both"]:
+                X_out = pd.concat([X_out, X_high], axis=1)
+        return X_out
 
     transform.__doc__ = BaseOutlier.transform.__doc__
```

Comment:
Also, instead of this, could we copy and paste the docstrings from the base outlier here so that we can add the bit about:

           **Note:** for 'Winsorizer()' with 'add_indicators=True', the
            output shape will be '[n_samples, 2*n_features]' or
            '[n_samples, 3*n_features]' depending on the tail mode.

Here and not in the base class? we want the docstrings to be specific and not communal :p